### PR TITLE
[jenkins]: add support for golang code coverage

### PR DESCRIPTION
**Pull request by Wayonb**
problem: no code coverage support for golang
solution: add golang checks for code coverage minimum

### CODE REVIEW

File: jenkins/shared-library/vars/codeCoverage.groovy [hunk 1]
```diff
@@ -43,6 +43,14 @@ void verifyCodeCoverageResult(String tool, Integer minimumCodeCoverage) {
 		'jacoco': { Integer target ->
 			logger.logInfo("Minimum code coverage set in pom is ${readJacocoCoverageLimit()}")
 			runScript('mvn jacoco:check@jacoco-check')
+		},
+		'golang': { Integer target ->
+			logCodeCoverageMinimum(target)
+			runScript('go tool cover -func coverage.out')
+			String coverage = runScript("go tool cover -func coverage.out | grep total | awk '{print substr(\$3, 1, length(\$3)-3)}'", true)
```
Comment (Jaguar0625) [inline jenkins/shared-library/vars/codeCoverage.groovy:50]: are you running coverage twice ?

Comment (Wayonb) [reply jenkins/shared-library/vars/codeCoverage.groovy:50]: This analyzes the coverage profiles generated by the test run to create coverage percentages.

Comment (Jaguar0625) [reply jenkins/shared-library/vars/codeCoverage.groovy:50]: i mean you are running script `go tool cover -func coverage.out` two times, is that intentional ?

Comment (Wayonb) [reply jenkins/shared-library/vars/codeCoverage.groovy:50]: yea, didnt want write code to parse the output 😅 

Will update 🙏🏾

File: jenkins/shared-library/vars/codeCoverage.groovy [hunk 2]
```diff
@@ -46,9 +52,10 @@ void verifyCodeCoverageResult(String tool, Integer minimumCodeCoverage) {
 		},
 		'golang': { Integer target ->
 			logCodeCoverageMinimum(target)
-			runScript('go tool cover -func coverage.out')
-			String coverage = runScript("go tool cover -func coverage.out | grep total | awk '{print substr(\$3, 1, length(\$3)-3)}'", true)
-			if (coverage.toInteger() < target) {
+			String coverageOutput = runScript('go tool cover -func coverage.out', true)
+			println(coverageOutput)
```
Comment (Jaguar0625) [inline jenkins/shared-library/vars/codeCoverage.groovy:56]: remove ?

**Review by Jaguar0625 — APPROVED** [review]
👍

File: jenkins/shared-library/vars/codeCoverage.groovy (hunk 2 above)
Comment (Wayonb) [reply jenkins/shared-library/vars/codeCoverage.groovy:56]: no, wanted to print the coverage results in the logs

**Pull request merged**